### PR TITLE
Fix NULL datetime for uimport/uexport

### LINKED ISSUE #6950: uimport doesn't support `"pwdreset_time":null`
### Expected behavior
`uimport` simply working.

### Actual behavior
`uimport` failing.

### Steps to reproduce the problem
1. export from Nerdica (I don't know what version they are using)
2. notice that the JSON file contains `"pwdreset_time":null`
3. fresh install
4. try `uimport`, it fails (with a red popup)
5. check the server side, logs, find:
    `uimport:insert user : ERROR : Incorrect datetime value: '' for
column 'pwdreset_time' at row 1 []`
6. substitute `"pwdreset_time":null` → `"pwdreset_time":"0001-01-01 00:00:00"`
7. retry `uimport`: it works!

### Friendica version you encountered the problem
854cc3e47296d4a1b01b893376064cf254d84f79

### Friendica source (git, zip)
git

### PHP version
7.3.3

### SQL version
MySQL 5.7.24

### PR DESCRIPTION
**Pull request by nupplaphil**
Fixes #6950 

- Tested uexport
- Tested uimport (with null and not null timestamp)

### CODE REVIEW

File: mod/uexport.php [hunk 1]
```diff
@@ -2,20 +2,26 @@
...
 use Friendica\App;
 use Friendica\Core\Hook;
 use Friendica\Core\L10n;
 use Friendica\Core\Renderer;
 use Friendica\Core\System;
 use Friendica\Database\DBA;
+use Friendica\Database\DBStructure;
 
 function uexport_init(App $a) {
+	global $dbStructure;
```
Comment (MrPetovan) [inline mod/uexport.php]: Please add a TODO comment about moving this global variable to a static Module variable when it is moved to `src/`.

File: src/Core/UserImport.php [hunk 2]
```diff
@@ -39,14 +39,21 @@ private static function checkCols($table, &$arr)
...
 		}
 		// remove inexistent columns
 		foreach ($arr as $icol => $ival) {
 			if (!in_array($icol, $tcols)) {
 				unset($arr[$icol]);
+				continue;
+			}
+
+			if ($ttype[$icol] === 'datetime') {
+				$arr[$icol] = !empty($ival) ? $ival : DBA::NULL_DATETIME;
```
Comment (MrPetovan) [inline src/Core/UserImport.php]:
```suggestion
				$arr[$icol] = $ival ?? DBA::NULL_DATETIME;
```

File: mod/uexport.php [hunk 3]
```diff
@@ -70,12 +88,25 @@ function _uexport_multirow($query) {
...
 	$result = [];
 	$r = q($query);
 	if (DBA::isResult($r)) {
+
 		foreach ($r as $rr) {
 			foreach ($rr as $k => $v) {
-				$result[$k] = $v;
+				switch ($dbStructure[$table]['fields'][$k]['type']) {
+					case 'datetime':
+						$result[$k] = !empty($v) ? $v : DBA::NULL_DATETIME;
```
Comment (MrPetovan) [inline mod/uexport.php]:
```suggestion
						$result[$k] = $v ?? DBA::NULL_DATETIME;
```

File: mod/uexport.php [hunk 4]
```diff
@@ -55,13 +61,25 @@ function uexport_content(App $a) {
...
 	$result = [];
 	$r = q($query);
 	if (DBA::isResult($r)) {
 		foreach ($r as $rr) {
 			$p = [];
 			foreach ($rr as $k => $v) {
-				$p[$k] = $v;
+				switch ($dbStructure[$table]['fields'][$k]['type']) {
+					case 'datetime':
+						$p[$k] = !empty($v) ? $v : DBA::NULL_DATETIME;
```
Comment (MrPetovan) [inline mod/uexport.php]:
```suggestion
						$p[$k] = $v ?? DBA::NULL_DATETIME;
```

**Comment by tobiasd** [comment]
Thanks!

**Pull request merged**